### PR TITLE
WT-6507 Exit cache eviction worker after our operation has timed out (v4.0 backport) 

### PR DESCRIPTION
**Pull request by tammybailey**

### CODE REVIEW

File: src/include/session.h [hunk 1]
```diff
@@ -62,6 +62,12 @@ struct __wt_session_impl {
     const char *lastop; /* Last operation */
     uint32_t id;        /* UID, offset in session array */
 
+    uint64_t operation_start_us;   /* Operation start */
+    uint64_t operation_timeout_us; /* Maximum operation period before rollback */
+#ifdef HAVE_DIAGNOSTIC
+    uint32_t op_5043_seconds; /* Temporary debugging to catch WT-5043, discard after 01/2020. */
```
Comment (luke-pearson) [inline src/include/session.h:68]: This shouldn't be needed.

Comment (tammybailey) [reply src/include/session.h]: Removed the diagnostic code here and in `src/include/time.i`.

File: dist/api_data.py [hunk 2]
```diff
@@ -1345,6 +1352,19 @@ def __cmp__(self, other):
         current transaction.  The value must also not be older than the
         current oldest and stable timestamps.  See
         @ref transaction_timestamps'''),
+    Config('durable_timestamp', '', r'''
```
Comment (kommiharibabu) [inline dist/api_data.py:1355]: I think there is no `durable_timestamp` support in MongoDB 4.0. This change shouldn't be backported?

Comment (tammybailey) [reply dist/api_data.py]: Removed, thanks Hari!

**Review by kommiharibabu — APPROVED** [review]
LGTM

**Review by luke-pearson — APPROVED** [review]
LGTM.

**Pull request merged**